src/main/js/ephox/imagetools/transformations/ImageTools.js

```diff
@@ -80,13 +82,13 @@ define("ephox/imagetools/transformations/ImageTools", [
 
   function resize(blob, w, h) {
     return Conversions.blobToImage(blob).then(function(image) {
-      var canvas = Canvas.create(w, h),
-        context = Canvas.get2dContext(canvas);
-
-      context.drawImage(image, 0, 0, w, h);
-      revokeImageUrl(image);
-
-      return Conversions.canvasToBlob(canvas, blob.type);
+      return ImageResizerWebgl.scale(image, w, h)
```

Comment:
Yeah, the last comment was overkill it should probably just state algorithms:
```
  ImageResizer.hermite,
  ImageResizer.gradient,
  ImageResizer.simple
]);
```

So it ties those in the priority order. Not sure if there is one that is preferred over the other but we just need to agree on the API here and then later on we can extend the list of resampling algoritmns and implementations of them.

So the above chooses the right alogoritm and technology based on platform and image size and maybe even image format since we can check if the image is a jpeg or a gif and do different things in theory.